tgui/packages/tgui/interfaces/GasAnalyzer.tsx

```diff
@@ -1,22 +1,47 @@
 import { useBackend } from '../backend';
 import { GasmixParser } from './common/GasmixParser';
 import type { Gasmix } from './common/GasmixParser';
+import { Button, Flex, LabeledList, Section, Stack, Box } from '../components';
 import { AtmosHandbookContent, atmosHandbookHooks } from './common/AtmosHandbook';
 import { Window } from '../layouts';
-import { Section } from '../components';
 
-export type GasAnalyzerData = {
+type GasmixHistory = {
+  allGasmixes: Gasmix[];
+};
+
+type GasAnalyzerData = {
   gasmixes: Gasmix[];
+  autoUpdating: boolean;
+};
+
+type GasAnalyzerHistoryData = {
+  historyGasmixes: GasmixHistory[];
+  historyViewMode: string;
+  historyIndex: number;
 };
 
 export const GasAnalyzerContent = (props, context) => {
   const { act, data } = useBackend<GasAnalyzerData>(context);
-  const { gasmixes } = data;
+  const { gasmixes, autoUpdating } = data;
```

Comment:
the backend does not always send data, so what happens when it tries to use .map on undefined?
```suggestion
  const { gasmixes = [], autoUpdating } = data;
```